src/compas_blender/artists/robotmodelartist.py

```diff
@@ -32,14 +37,49 @@ def __init__(self,
                  **kwargs: Any):
         super().__init__(model=model, collection=collection or model.name, **kwargs)
 
+    # this method should not be here
+    # it has nothing to do with the current object
     def transform(self, native_mesh: bpy.types.Object, transformation: Transformation) -> None:
+        """Transform the mesh of a robot model.
+
+        Parameters
+        ----------
+        native_mesh : bpy.types.Object
+            A mesh scene object.
+        transformation : :class:`compas.geometry.Transformation`
+            A transformation matrix.
+
+        Returns
+        -------
+        None
+
+        """
         native_mesh.matrix_world = mathutils.Matrix(transformation.matrix) @ native_mesh.matrix_world
 
+    # again
+    # doesn't make sense to me that there is no reference to self (except for the collection)
+    # suggests that this method shouldn't be here
     def create_geometry(self,
-                        geometry: Union[Mesh, Shape],
+                        geometry: Union[Mesh, Shape],  # seems incorrect to also accept a shape since it does not have an attribute dict by default
```

Comment:
you're right. the type hint is incorrect, this only works with meshes.
```suggestion
                        geometry: Mesh,
```